src/linter/side_effects_finder.go

```diff
@@ -99,12 +100,11 @@ var pureBuiltins = func() map[string]struct{} {
 }()
 
 func (f *sideEffectsFinder) functionCallIsPure(n *expr.FunctionCall) bool {
-	// Allow referencing builtin funcs even before indexing is completed.
-	nm, ok := n.Function.(*name.Name)
-	var funcName string
-	if ok && len(nm.Parts) == 1 {
+	// We allow referencing builtin funcs even before indexing is completed.
+
+	funcName, ok := solver.GetFuncName(f.st, n.Function)
```

Comment:
Won't it cause a race condition? You are updating the index during indexing, you can't use it to resolve anything yet.

Reply:
You're right.

Reply:
Fixed.